instagram/src/App.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component } from "react";
+import "./App.css";
+import PostPage from "./Components/PostContainer/PostPage";
+import SearchBar from "./Components/SearchBar/SearchBar";
+import Authenticate from "./Components/Authentication/Authenticate";
+
+const PostPageAuthed = Authenticate(PostPage);
+
+class App extends Component {
+  state = {
+    username: "",
+    password: "",
+    credentials: localStorage.getItem("credentials")
+  };
+
+  handleCreds = credentials => {
+    if (!credentials) localStorage.removeItem("credentials");
+    this.setState({ credentials });
+  };
+
+  handleChange = event => {
+    this.setState({ [event.target.name]: event.target.value });
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
+    const authKey = `${this.state.username} is logged in`;
+    localStorage.setItem("credentials", authKey);
+    this.setState({ username: "", password: "" });
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <SearchBar handleCreds={this.handleCreds} />
+        <PostPageAuthed
+          credentials={this.state.credentials}
+          username={this.state.username}
```

Comment:
I think username isn't being set correctly. From reading the code i bet it's always an empty string when passed down. same for password.